columnphysics/icepack_shortwave.F90

```diff
@@ -2710,20 +2711,18 @@ subroutine compute_dEdd (nilyr,    nslyr,    klev,  klevp,  &
                   endif       ! modal_aero
 !mgf--
                   enddo       ! na
-                  gaer = gaer/(waer+puny)
-                  waer = waer/(taer+puny)
-                  g(k)   = (g(k)*w0(k)*tau(k) + gaer*waer*taer) / &
-                           (w0(k)*tau(k) + waer*taer)
-                  w0(k)  = (w0(k)*tau(k) + waer*taer) / &
-                           (tau(k) + taer)
+                  g(k)   = (g(k)*w0(k)*tau(k) + gaer) / &
+                                (w0(k)*tau(k) + waer)
+                  w0(k)  =      (w0(k)*tau(k) + waer) / &
+                                      (tau(k) + taer)
                   tau(k) = tau(k) + taer
                enddo       ! k
             endif     ! tr_aero
 
             ! pond
          else !if( srftyp == 2 ) then
             ! pond water layers evenly spaced
-            dz = hp/(c1/rnslyr+c1)
+            dz = hp/(real(nslyr,kind=dbl_kind)+c1)
```

Comment:
I think this change is correct, but I question the original expression -- does anyone understand why the +1 is in the denominator?  It looks like the dEdd scheme is dividing  the pond depth (volume) evenly among the snow layers, but then why is this not `dz=hp/nslyr` ?

Reply:
I believe this was the "surface scattering layer" versus the "internal layer". So, I think this is fine.

Reply:
Okay, now I see that the loop immediately following this line is over k = 0, nslyr.  So the ponds are simply divided into nslyr+1 equally-spaced layers for the dEdd calculations, and I won't worry about what "SSL" might mean for a pond -- it's just bookkeeping for consistency with the snow calculation, I guess.  Thanks.